tests/integration/API/JSON/v1/SegmentsTest.php

```diff
@@ -162,6 +163,62 @@ public function testItReturnsErrorWhenTrashingSegmentWithActiveNewsletter() {
     expect($response->errors[0]['message'])->equals("List cannot be deleted because it’s used for 'Subject' email");
   }
 
+  public function testItReturnsErrorWhenTrashingSegmentWithActiveForm() {
+    $form = new FormEntity('My Form');
+    $form->setSettings([
+      'segments' => [(string)$this->segment3->getId()],
+    ]);
+    $this->entityManager->persist($form);
+    $this->entityManager->flush();
+
+    $response = $this->endpoint->trash(['id' => $this->segment3->getId()]);
+    $this->entityManager->refresh($this->segment3);
+    expect($response->status)->equals(APIResponse::STATUS_BAD_REQUEST);
+    expect($response->errors[0]['message'])->equals("List cannot be deleted because it’s used for 'My Form' form");
+  }
+
+  public function testItReturnsPluralErrorWhenTrashingSegmentWithActiveForms() {
+    $form1 = new FormEntity('My Form');
+    $form1->setSettings([
+      'segments' => [(string)$this->segment3->getId()],
+    ]);
+    $this->entityManager->persist($form1);
+    $this->entityManager->flush();
+
+    $form2 = new FormEntity('My other Form');
+    $form2->setSettings([
+      'segments' => [(string)$this->segment3->getId()],
+    ]);
+    $this->entityManager->persist($form2);
+    $this->entityManager->flush();
+
+    $response = $this->endpoint->trash(['id' => $this->segment3->getId()]);
+    $this->entityManager->refresh($this->segment3);
+    expect($response->status)->equals(APIResponse::STATUS_BAD_REQUEST);
+    expect($response->errors[0]['message'])->equals("List cannot be deleted because it’s used for 'My Form', 'My other Form' forms");
+  }
+
+  public function testItCanTrashSegmentWithoutActiveForm() {
+    $form1 = new FormEntity('My Form');
+    $form1->setSettings([
+      'segments' => [(string)$this->segment3->getId()],
+    ]);
+    $this->entityManager->persist($form1);
+    $this->entityManager->flush();
+
+    $response = $this->endpoint->trash(['id' => $this->segment2->getId()]);
+    $this->entityManager->clear();
```

Comment:
Here comes another question from someone that is still learning Doctrine. Why do we need to call `clear()` here?

Reply:
And another answer from someone also not familiar with Doctrine:
`clear` would force reloading all the entities and it was not needed there. Changed it to refresh only the segment.

Reply:
Resolving this one as it is very similar to the conversation above (https://github.com/mailpoet/mailpoet/pull/3605#discussion_r680359953)